airbyte-cdk/python/unit_tests/sources/test_abstract_source.py

```diff
@@ -254,11 +289,24 @@ def _state(state_data: Dict[str, Any]):
 
 
 class TestIncrementalRead:
-    def test_with_state_attribute(self, mocker):
+    @pytest.mark.parametrize("use_legacy", [True, False])
```

Comment:
nit: the input state could be parametrized, and adding an id to the pytest param would help understand what this param refers to (legacy / stream_state)